### ISSUE TITLE
gtk-0.15.8 fails to build with GHC 9.8

### ISSUE THREAD

**Issue by andreasabel** [gtk2hs/gtk2hs#332]
With `--allow-newer={text,bytestring}` one can try to build `gtk-0.15.8` (latest Hackage release) with GHC 9.8, but I get build errors:
```
Graphics/UI/Gtk/Embedding/Socket.chs:175:6: error: 
    • Couldn't match expected type: Ptr ()
                  with actual type: Maybe DrawWindow
    • In the second argument of ‘\ (Socket arg1) arg2
                                   -> withForeignPtr arg1
                                        $ \ argPtr1 -> gtk_socket_add_id argPtr1 arg2’, namely
        ‘(fromNativeWindowId windowId)’
      In the expression:
        (\ (Socket arg1) arg2
           -> withForeignPtr arg1
                $ \ argPtr1 -> gtk_socket_add_id argPtr1 arg2)
          (toSocket self) (fromNativeWindowId windowId)
    |
175 |     (fromNativeWindowId windowId)
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
etc.

**Comment by Bodigrim** [comment]
I can confirm: git repo is buildable on macOS, but the released `gtk-0.15.8` is not. Strange, I don't see what has changed since 8cedf368.

**Comment by Bodigrim** [comment]
That said, `gtk-0.15.8` fails to build with GHC 9.6 on my macOS machine as well, so this is not related to GHC 9.8.

On the other hand `gtk-0.15.8` + GHC 9.8 + Ubuntu compiles fine as evidenced by https://github.com/def-/ghc-vis/actions/runs/7441387274/job/20243429113. 

So I think it's fair to revise `gtk-0.15.8` to allow GHC 9.8 boot libraries.

**Comment by andreasabel** [comment]
> So I think it's fair to revise `gtk-0.15.8` to allow GHC 9.8 boot libraries.

@Bodigrim Yes, if you convinced yourself that it works on Ubuntu I think you should go ahead and relax the bounds.

**Comment by kidq330** [comment]
M3 macOS 14.4.1 here, also had more luck with the PR version

**Comment by hamishmack** [comment]
```
Graphics/UI/Gtk/Embedding/Socket.chs:175:6: error: 
    • Couldn't match expected type: Ptr ()
                  with actual type: Maybe DrawWindow
```

This error is what you get if `have-quartz-gtk` is not set when it needs to be.

> Cabal flag -f have-quartz-gtk did not help

I just had this same problem.  It looks like setting the flag does not rerun the CPP on `chs` files.  I can reproduce the problem you are seeing by first building without the flag (getting the error).  Then setting the flag and building again.  Even though the flag is not set I still see the error.

Running `rm -rf dist-newstyle` to force a rebuild solves it.
